larpandoracontent/LArTwoDReco/HitWidthClusterMergingAlgorithm.cc

```diff
@@ -0,0 +1,363 @@
+/**
+ *  @file   larpandoracontent/LArTwoDReco/HitWidthClusterMergingAlgorithm.cc
+ *
+ *  @brief  Implementation of the hit width cluster merging algorithm class.
+ *
+ *  $Log: $
+ */
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "larpandoracontent/LArTwoDReco/HitWidthClusterMergingAlgorithm.h"
+#include "larpandoracontent/LArHelpers/LArClusterHelper.h"
+#include "larpandoracontent/LArHelpers/LArGeometryHelper.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+HitWidthClusterMergingAlgorithm::HitWidthClusterMergingAlgorithm() :
+  m_maxConstituentHitWidth(0.5),
+  m_hitWidthScalingFactor(1.0),
+  m_fittingWeight(10),
+  m_minClusterWeight(0.5),      
+  m_maxXMergeDistance(5),     
+  m_maxZMergeDistance(2),     
+  m_minMergeCosOpeningAngle(0.97),
+  m_minDirectionDeviationCosAngle(0.90)
+{
+}
+    
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void HitWidthClusterMergingAlgorithm::GetListOfCleanClusters(const ClusterList *const pClusterList, ClusterVector &clusterVector) const
+{
+    // clear map if already full i.e. from other view clustering
+    if (!m_clusterToParametersMap.empty())
```

Comment:
Here the cluster vector is first determined which contains all the clusters that I want to break apart (i.e. make cluster parameter objects out of). I thought it was best to fill the m_clusterToParametersMap map here since it saves me looping over the clusterVector again at the start of the PopulateClusterAssociationMap function (which directly follows GetListOfCleanClusters in function calls).